### PR TITLE
NATIVE-43: Fetch data instead of using mock data

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

File: src/routes/extras/containers/ExtrasContainer.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import type { ThemeType } from '../../../modules/theme/constants/theme'
+import request from '../../../modules/endpoint/request'
+import type { StateType } from '../../../modules/app/StateType'
+import type { NavigationScreenProp } from 'react-navigation'
+import { baseUrl } from '../../../modules/endpoint/constants'
+
+const mapStateToProps = (state: StateType, ownProps) => {
+  const language = state.cityContent.language
+  if (!language) {
+    throw new Error('Language not set for categories!')
```
Comment (klinzo) [inline src/routes/extras/containers/ExtrasContainer.js]: Change to `extras`

File: src/routes/extras/containers/ExtrasContainer.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  navigateToExtra = (path: string, isExternalUrl: boolean, offerHash: ?string = null) => {
+    if (isExternalUrl) {
+      Linking.openURL(path)
+    } else if (this.props.navigation.push) {
+      const params = {city: this.props.city, extras: this.state.extras, offerHash: offerHash}
+      this.props.navigation.push(path, params)
+    }
+  }
+
+  async loadExtras () {
```
Comment (klinzo) [inline src/routes/extras/containers/ExtrasContainer.js]: Is it on purpose to have this loading/fetching code here? This isn't handled the same way for events/categories, is it?

Comment (maxammann) [reply src/routes/extras/containers/ExtrasContainer.js]: Extras are only available when you are online. That's why I decided not to put it in the store. Very simple to do it like this.

File: src/routes/wohnen/containers/WohnenExtraContainer.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    return <WohnenExtra wohnenExtra={extra} offerHash={offerHash} navigateToOffer={navigateToOffer} offers={offers}
+                        t={t} />
+  }
+}
+
+export default compose(
+  connect(mapStateToProps),
+  translate('wohnen'),
+  translate()
```
Comment (klinzo) [inline src/routes/wohnen/containers/WohnenExtraContainer.js]: Remove

File: src/routes/wohnen/containers/WohnenExtraContainer.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  render () {
+    const {extra, offerHash, navigateToOffer, t} = this.props
+    const {offers, error} = this.state
+
+    if (error) {
+      return <Failure error={error} />
+    }
+
+    if (!offers) {
+      return <ActivityIndicator size='large' color='#0000ff' />
```
[the remark this answers is not in the document]
Comment (maxammann) [reply src/routes/wohnen/containers/WohnenExtraContainer.js]: NATIVE-211

File: src/routes/sprungbrett/containers/SprungbrettExtraContainer.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  render () {
+    const {extra, t} = this.props
+    const {jobs, error} = this.state
+
+    if (error) {
+      return <Failure error={error} />
+    }
+
+    if (!jobs) {
+      return <ActivityIndicator size='large' color='#0000ff' />
```
[the remark this answers is not in the document]
Comment (maxammann) [reply src/routes/sprungbrett/containers/SprungbrettExtraContainer.js]: https://issues.integreat-app.de/browse/NATIVE-211

File: src/routes/extras/containers/ExtrasContainer.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  render () {
+    const {theme, t} = this.props
+    const {extras, error} = this.state
+
+    if (error) {
+      return error.message
+    }
+
+    if (!extras) {
+      return <ActivityIndicator size='large' color='#0000ff' />
```
[the remark this answers is not in the document]
Comment (maxammann) [reply src/routes/extras/containers/ExtrasContainer.js]: https://issues.integreat-app.de/browse/NATIVE-211

File: src/routes/extras/containers/ExtrasContainer.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    this.state = {extras: null, error: null}
+  }
+
+  componentWillMount () {
+    this.loadExtras()
+  }
+
+  navigateToExtra = (path: string, isExternalUrl: boolean, offerHash: ?string = null) => {
+    if (isExternalUrl) {
+      Linking.openURL(path)
```
[the remark this answers is not in the document]
Comment (maxammann) [reply src/routes/extras/containers/ExtrasContainer.js]: https://issues.integreat-app.de/browse/NATIVE-211

**Pull request merged**